adapter/main.go

```diff
@@ -93,45 +92,39 @@ func (a *Adapter) makeProvider(ctx context.Context, globalHTTPTimeout time.Durat
 		return nil, nil, fmt.Errorf("failed to add keda scheme to runtime scheme (%s)", err)
 	}
 
-	kubeclient, err := client.New(cfg, client.Options{
-		Scheme: scheme,
+	namespace, err := getWatchNamespace()
+	if err != nil {
+		logger.Error(err, "failed to get watch namespace")
+		return nil, nil, fmt.Errorf("failed to get watch namespace (%s)", err)
+	}
+	mgr, err := ctrl.NewManager(ctrl.GetConfigOrDie(), ctrl.Options{
+		Scheme:    scheme,
+		Namespace: namespace,
 	})
 	if err != nil {
-		logger.Error(err, "unable to construct new client")
-		return nil, nil, fmt.Errorf("unable to construct new client (%s)", err)
+		logger.Error(err, "failed to setup manager")
+		return nil, nil, err
 	}
 
 	broadcaster := record.NewBroadcaster()
 	recorder := broadcaster.NewRecorder(scheme, corev1.EventSource{Component: "keda-metrics-adapter"})
-	handler := scaling.NewScaleHandler(kubeclient, nil, scheme, globalHTTPTimeout, recorder)
+	handler := scaling.NewScaleHandler(mgr.GetClient(), nil, scheme, globalHTTPTimeout, recorder)
 	externalMetricsInfo := &[]provider.ExternalMetricInfo{}
 	externalMetricsInfoLock := &sync.RWMutex{}
 
-	namespace, err := getWatchNamespace()
-	if err != nil {
-		logger.Error(err, "failed to get watch namespace")
-		return nil, nil, fmt.Errorf("failed to get watch namespace (%s)", err)
-	}
-
 	prometheusServer := &prommetrics.PrometheusMetricServer{}
 	go func() { prometheusServer.NewServer(fmt.Sprintf(":%v", prometheusMetricsPort), prometheusMetricsPath) }()
 	stopCh := make(chan struct{})
-	if err := runScaledObjectController(ctx, scheme, namespace, handler, logger, externalMetricsInfo, externalMetricsInfoLock, maxConcurrentReconciles, stopCh); err != nil {
+
+
```

Comment:
Please, delete one of these 2 empty lines

Reply:
updated